src/schema/v2/me/artworkRecommendations.ts

```diff
@@ -0,0 +1,82 @@
+import { GraphQLFieldConfig, GraphQLInt } from "graphql"
+import { connectionFromArray } from "graphql-relay"
+import gql from "lib/gql"
+import { convertConnectionArgsToGravityArgs, extractNodes } from "lib/helpers"
+import { compact, find } from "lodash"
+import { CursorPageable, pageable } from "relay-cursor-paging"
+import { createPageCursors } from "schema/v1/fields/pagination"
+import { artworkConnection } from "schema/v2/artwork"
+import { ResolverContext } from "types/graphql"
+
+// This limits the maximum number of artworks we receive from Vortex and is related to how we implement the Connection in this resolver.
+const MAX_ARTWORKS = 50
+
+export const ArtworkRecommendations: GraphQLFieldConfig<
+  void,
+  ResolverContext
+> = {
+  description: "A connection of artwork recommendations for the current user.",
+  type: artworkConnection.connectionType,
+  args: pageable({
+    page: { type: GraphQLInt },
+  }),
+  resolve: async (
+    _root,
+    args: CursorPageable,
+    { vortexGraphqlLoader, artworksLoader }
+  ) => {
+    if (!vortexGraphqlLoader || !artworksLoader) return
+
+    const { page, size } = convertConnectionArgsToGravityArgs(args)
+
+    // Fetch artwork IDs from Vortex
+
+    const vortexResult = await vortexGraphqlLoader({
+      query: gql`
+        query artworkRecommendationsQuery {
+          artworkRecommendations(first: ${MAX_ARTWORKS}) {
+            totalCount
+            edges {
+              node {
+                artworkId
+                score
+              }
+            }
+          }
+        }
+      `,
+    })()
+
+    const artworkRecommendations = extractNodes(
+      vortexResult.data?.artworkRecommendations
+    )
+    const artworkIds = artworkRecommendations.map(
+      (node: any) => node?.artworkId
+    )
+
+    // Fetch artwork details from Gravity
+
+    let artworks: any[] = []
+
+    if (artworkIds?.length) {
+      artworks = await artworksLoader({
+        ids: artworkIds,
+      })
+
+      // Apply order from Vortex result (score ASC)
```

Comment:
what do you think about adding a test for this specific behaviour?

Reply:
Good catch. The test case implicitly tested the order. I've added it to the description of the test to make it more clear.